packages/commonwealth/client/scripts/views/components/AddToHomeScreenPrompt/AndroidPrompt/AndroidPrompt.tsx

```diff
@@ -20,6 +20,7 @@ export const AndroidPrompt = ({
   const { animationStyles } = useAnimation({ transitionDuration: '.5s' });
   let installPromptEvent = null;
   const [checkboxChecked, setCheckboxChecked] = useState(false);
+  const [showInstallButton, setShowInstallButton] = useState(true);
```

Comment:
i think the original issue 
> Clicking PWA install prompt on Android gives TypeError: Cannot read properties of null (reading 'prompt') at onClick.

is because, in some scenarios `let installPromptEvent = null;` gets reset whenever the component re-renders, and `installPromptEvent.prompt();` doesn't work, while in other scenarios the `window.addEventListener('beforeinstallprompt` doesn't work for specific devices. 

Maybe we should make `installPromptEvent` a ref, unless we originally (when PWA was added) wanted to reset this state when component re-renders?

Reply:
@mzparacha Yes, I can try this. Thanks for the suggestion.

Reply:
@mzparacha Just tried implementing a useRef in the way that you suggested and it still didn't work to install the pwa from clicking Install. In Slack this morning Marcin mentioned that he and Tim tried to get this to work and couldn't, so I think for this PR this is as far as it should go. If we want to devote more time to trying to get this to work with an addEventListener, a new bug ticket can be made.

Reply:
you could also do this:

useCallback(() => {
  return () => {
    console.log("I've been unmounted");
  }
}, []);

which will debug an unmount for you to see if that's happening.